xbmc/utils/GUIUtils.cpp

```diff
@@ -0,0 +1,363 @@
+/*
+ *  Copyright (C) 2016-2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "GUIUtils.h"
+
+#include "Application.h"
+#include "GUIPassword.h"
+#include "PartyModeManager.h"
+#include "PlayListPlayer.h"
+#include "ServiceBroker.h"
+#include "dialogs/GUIDialogBusy.h"
+#include "dialogs/GUIDialogKaiToast.h"
+#include "dialogs/GUIDialogYesNo.h"
+#include "filesystem/File.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "guilib/LocalizeStrings.h"
+#include "messaging/helpers/DialogHelper.h"
+#include "messaging/helpers/DialogOKHelper.h"
+#include "music/tags/MusicInfoTag.h"
+#include "playlists/PlayList.h"
+#include "playlists/PlayListFactory.h"
+#include "profiles/ProfileManager.h"
+#include "settings/SettingsComponent.h"
+#include "utils/log.h"
+#include "view/GUIViewState.h"
+
+using namespace KODI::MESSAGING;
+
+using namespace XFILE;
+using namespace std::chrono_literals;
+
+using KODI::MESSAGING::HELPERS::DialogResponse;
+
+// set and reset in worker thread, read by calling thread
+CEvent CGUIUtils::m_utilsJobEvent(false, false);
+
+class CAddItemToPlaylistJob : public CJob
+{
+  const CFileItemPtr& p_Item;
+  CFileItemList& queued_Items;
+
+public:
+  CAddItemToPlaylistJob(const CFileItemPtr& pItem, CFileItemList& queuedItems)
+    : p_Item(pItem), queued_Items(queuedItems)
+  {
+  }
+
+  ~CAddItemToPlaylistJob(void) override = default;
+
+  bool DoWork(void) override
```

Comment:
You should remove the `void`. It's simply not needed and we don't do this usually in our code base.